src/pkg/packager/validator/validator.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package validator contains functions for linting the zarf.yaml
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	zarfInvalidPrefix = "schema is invalid:"
+	zarfWarningPrefix = "zarf schema warning:"
+)
+
+// ValidateZarfSchema a zarf file against the zarf schema, returns an error if the file is invalid
+func ValidateZarfSchema(path string) (err error) {
+	var zarfTypedData types.ZarfPackage
+	if err := utils.ReadYaml(filepath.Join(path, layout.ZarfYAML), &zarfTypedData); err != nil {
+		return err
+	}
+
+	if err := checkForVarInComponentImport(zarfTypedData); err != nil {
+		message.Warn(err.Error())
+	}
+
+	zarfSchema, _ := config.GetSchemaFile()
+
+	var zarfData interface{}
+	if err := utils.ReadYaml(filepath.Join(path, layout.ZarfYAML), &zarfData); err != nil {
+		return err
+	}
+
+	if err = validateSchema(zarfData, zarfSchema); err != nil {
+		return err
+	}
+
+	message.Success("Validation successful")
+	return nil
+}
+
+func checkForVarInComponentImport(zarfYaml types.ZarfPackage) error {
```

Comment:
this logic is a bit complicated for concatting multiple errors. chatgpt came up with this:

func checkForVarInComponentImport(zarfYaml types.ZarfPackage) error {
	var errorMessages []string

	for i, component := range zarfYaml.Components {
		if strings.Contains(component.Import.Path, types.ZarfPackageTemplatePrefix) {
			errorMessages = append(errorMessages, fmt.Sprintf("component.%d.import.path will not resolve ZARF_PKG_TMPL_* variables", i))
		}
		if strings.Contains(component.Import.URL, types.ZarfPackageTemplatePrefix) {
			errorMessages = append(errorMessages, fmt.Sprintf("component.%d.import.url will not resolve ZARF_PKG_TMPL_* variables", i))
		}
	}

	if len(errorMessages) > 0 {
		return errors.New(zarfWarningPrefix + strings.Join(errorMessages, ", "))
	}
	return nil
}

Reply:
I like this, thanks Barry!